README.md

```diff
@@ -108,37 +108,100 @@ $ aut account balance 0x4EcE2e62E67a7B64a83D3E180dC86962145b762f
+
+```console
+$ aut  contract call --abi $(aut protocol contract-abi-path) --address $(aut protocol contract-address) \
+  getProposer 10000 1 
+"0x31870f96212787D181B3B2771F58AF2BeD0019Aa"
+```
+
+#### Complex types
 
-The [autonity.py](https://github.com/autonity/autonity.py) dependency
-is included as a submodule, for ease of development.  Sync all submodules, e.g.:
+The types `array` and `tuple` are supported as parameters as single quoted strings.
+
+For example, giving a contract function `arrayLength` that takes a `string[]` input, the command will look like the following: 
 ```console
-$ git submodule update --init --recursive
+$ aut contract call --abi ABI_FILE_PATH --address CONTRACT_ADDRESS \
+  arrayLength '["a","b","c","d","e","f","g"]'
+7
+```
+
+The same applies to `tuple` types, giving a struct:
+```solidity
+struct Payment {
+    address from; 
+    uint256 blockNumber;
+    uint256 amount;  
+}
 ```
+A contract call `printHeight` that accepts a `Payment` as input will be:
 
-Create and activate a virtual-env for development:
 ```console
-$ python -m venv env
-$ . env/bin/activate
+$ aut contract call --abi ABI_FILE_PATH --address CONTRACT_ADDRESS  \
+  printHeight '["0x31870f96212787D181B3B2771F58AF2BeD0019Aa", 183413, 1000000000000000000]'
+183413
 ```
 
-(Note the `(env)` prefix to your prompt.  Activate the venv from other
-terminals using the same `. env/bin/activate` command.)
+## Development
+
+The project is managed using [hatch](https://hatch.pypa.io/latest/). Check the installation instructions [here](https://hatch.pypa.io/latest/install/).
+
+Hatch will automatically manage a virtual environment for the project. To run the command in development mode use:
 
-Install both `autetl` and `autonity.py` in "editable" mode, within this virtual-env:
-development:
 ```console
-(env)$ make
+$ hatch run aut ....
 ```
 
-(Note that the `aut` command is also installed in the virtual-env.)
+alternatively, you can open a shell in the `hatch` environment by executing 
+
+```console
+$ hatch shell
+```
 
 To run all code checks (linters, type-checker, unit tests, etc):
 ```console
-(env)$ make check
+$ make check
 ```
 
-Several tests scripts (which invoke the `aut` command itself) are available in the [scripts](./scripts) directory.  The are intended to be run within the Python virtual-env, from the repository root directory.
+Several test scripts (which invoke the `aut` command itself) are available in the [scripts](./scripts) directory.  They are intended to be run within the Python virtual-env, from the repository root directory.
+
+#### Autonity.py
+
+The `aut` CLI depends on the [`autonity.py`](https://github.com/autonity/autonity.py) package, to develop using a local version of the `autonity.py` package you need to update the `pyproject.toml`:
+
+For the `project.dependency` set the path to the local `autonity.py` project:
```

Comment:
Just a suggestion to make it super obvious that people can skip this section and not worry about it unless they need to develop against a custom version of `autonity.py`...

```suggestion
The `aut` CLI depends on the [`autonity.py`](https://github.com/autonity/autonity.py) package, and will pull in the right version as specified in `pyproject.toml`.

The instructions in this section are only required if you need to develop against a local version of the `autonity.py` package.

To support this workflow, you need to update the `pyproject.toml` as follows. For the `project.dependency` set the path to the local `autonity.py` project:
```